tests/Feature/Http/Controllers/EntryControllerTest.php

```diff
@@ -227,6 +227,21 @@ public function update_redirects()
         $entry->user()->associate($this->user)->save();
         $entry->character()->associate($character)->save();
 
+        $response = $this->actingAs($this->user)->put(route('entry.update', $entry), [
+            'adventure_id' => $adventure->id,
+            'campaign_id' => $campaign->id,
+            'event_id' => $event->id,
+            'dungeon_master_id' => $dungeon_master_user->id,
+            'dungeon_master' => $dungeon_master,
+            'date_played' => $date_played,
+            'location' => $location,
+            'type' => $type,
+            'levels' => $levels,
+            'gp' => $gp,
+        ]);
+
+        $response->assertForbidden();
```

Comment:
I don't think I understand why you're expecting this response to fail. Doesn't the entry update keep the last character attached to it?

Reply:
There's no character_id in the *request body* and type is not 'dm' so it fails authorization in EntryUpdateRequest.

Reply:
Got it, thanks.